sycl/doc/design/CommandGraph.md

```diff
@@ -166,7 +166,12 @@ created on UR command-buffer enqueue.
 There is also a *WaitEvent* used by the `ur_exp_command_buffer_handle_t` class
 in the prefix to wait on any dependencies passed in the enqueue wait-list.
 
-![L0 command-buffer diagram](images/L0_UR_command-buffer.svg)
+If a command-buffer is about to be submitted to a queue with the profiling 
+property enabled, a profiling *StartEvent* is added to the end of the prefix.
+This event is used to get the command-buffer execution start time on the 
+device. 
```

Comment:
Forgive me if this has already been discussed elsewhere, is there a reason we can't add the start event to the beginning of the actual graph command list on command buffer creation? I'd guess because we need access to the queue to associate the event, but maybe I'm wrong.

We're introducing some inaccuracy here I would think , although I don't know how big or how static that extra time is, and maybe there's no way around it.

Reply:
I agree on the accuracy loss, but we need an unique event for a submission.
Indeed, since a command-buffer can be submitted multiple times, if we rely on the same event for each submission (i.e. an event associated with the command-buffer) the previous profiling info will be override by subsequent submissions.
And, we cannot add a submission unique event to the actual graph command list since this list is closed during the finalize step.

I would add that the accuracy issue is even double, since the end time is also taken for an event which belongs to the suffix command-list which may be not executed as soon as the actual graph command-list is complete.